APMrover2/mode_smart_rtl.cpp

```diff
@@ -52,29 +57,27 @@ void ModeSmartRTL::update()
                 }
                 _load_point = false;
                 // set target destination to new point
-                if (!set_desired_location_NED(next_point)) {
+                if (!g2.wp_nav.set_desired_location_NED(next_point)) {
                     // this failure should never happen but we add it just in case
                     gcs().send_text(MAV_SEVERITY_INFO, "SmartRTL: failed to set destination");
                     smart_rtl_state = SmartRTL_Failure;
```

Comment:
This is pre-existing, but wouldn't we want to break here?

Reply:
I'm not sure breaking would be good.  It would mean that for that one iteration the vehicle would be uncontroller - it would keep the throttle and steering at the values from the previous iteration.  So then there would be some small risk that it keeps hitting that code path over and over and we end up with a drive away..

Reply:
I think I understand your point, although it would be the same case above of reaching home (except here we just reached one point and couldn't set the next one) and in that case it has a break - which I think is there to prevent the code here to run, but it does have the effect you are talking about.